server/src/main/java/org/opensearch/tasks/Task.java

```diff
@@ -108,19 +122,41 @@ public Task(
      *            generate data?
      */
     public final TaskInfo taskInfo(String localNodeId, boolean detailed) {
+        return taskInfo(localNodeId, detailed, detailed);
+    }
+
+    /**
+     * Build a version of the task status you can throw over the wire and back
+     * with the option to include resource stats or not.
+     * This method is only used during creating TaskResult to avoid storing resource information into the task index.
+     *
+     * @param includeStats should information include resource stats.
+     *                     By default, detailed flag is used to control including resource information.
+     *                     But inorder to avoid storing resource stats into task index as strict mapping is enforced and breaks when adding this field.
+     *                     In the future, task-index-mapping.json can be modified to add resource stats.
+     */
+    private TaskInfo taskInfo(String localNodeId, boolean detailed, boolean includeStats) {
```

Comment:
- details already includes stats. So this ideally should be `excludeStats` if we want to support it for now

Reply:
Modified.